ext/intl/uchar/uchar.c

```diff
@@ -536,9 +553,15 @@ IC_METHOD(getFC_NFKC_Closure) {
 /* {{{ proto bool IntlChar::<name>(int|string $codepoint) */
 #define IC_BOOL_METHOD_CHAR(name) \
 IC_METHOD(name) { \
-	UChar32 cp; zval *zcp; \
-	if ((zend_parse_parameters(ZEND_NUM_ARGS(), "z", &zcp) == FAILURE) || \
-	    (convert_cp(&cp, zcp) == FAILURE)) { return; } \
+	UChar32 cp; \
+	zend_string *string_codepoint; \
+	zend_long int_codepoint; \
+	ZEND_PARSE_PARAMETERS_START(1, 1) \
+		Z_PARAM_STR_OR_LONG(string_codepoint, int_codepoint) \
+	ZEND_PARSE_PARAMETERS_END(); \
```

Comment:
To avoid code-size increase, let's move this common parameter parsing code into a function. Maybe something like

```
static zend_never_inline int parse_code_point_param(UChar32 *cp) {
    zend_string *string_codepoint;
    zend_long int_codepoint;
    ZEND_PARSE_PARAMETERS_START(1, 1)
        Z_PARAM_STR_OR_LONG(string_codepoint, int_codepoint)
    ZEND_PARSE_PARAMETERS_END_EX(return FAILURE);
    return convert_cp(cp, string_codepoint, int_codepoint);
}
```

Will also simplify the code in places.